docs/framework/magento2.md

```diff
@@ -1 +1,214 @@
+        </arguments>
+    </type>
+</config>
+```
+
+## Example Transport
+```php
+<?php
+declare(strict_types=1);
+
+namespace Mage\OpenWeather\Http;
+
+use Phpro\HttpTools\Client\Factory\LazyClientLoader;
+use Phpro\HttpTools\Request\RequestInterface;
+use Phpro\HttpTools\Transport\Json\JsonTransport;
+use Phpro\HttpTools\Transport\TransportInterface;
+use Phpro\HttpTools\Uri\TemplatedUriBuilder;
+
+class Transport implements TransportInterface
+{
+    /**
+     * @var TransportInterface
+     */
+    private $jsonTransport;
+
+    public function __construct(
+        LazyClientLoader $clientLoader,
+        TemplatedUriBuilder $uriBuilder
+    ) {
+        $this->jsonTransport = JsonTransport::createWithAutodiscoveredPsrFactories(
+            $clientLoader->load(),
+            $uriBuilder
+        );
+    }
+
+    public function __invoke(RequestInterface $request): array
+    {
+        return ($this->jsonTransport)($request);
+    }
+}
+```
+
+## Example Request Handler
+```php
+<?php
+declare(strict_types=1);
+
+namespace Mage\OpenWeather\Http\RequestHandler;
+
+use Mage\OpenWeather\Http\Request\WeatherRequest;
+use Mage\OpenWeather\Http\Response\WeatherResponse;
+use Phpro\HttpTools\Transport\TransportInterface;
+
+class Weather
+{
+    /**
+     * @var TransportInterface
+     */
+    private $transport;
+
+    public function __construct(TransportInterface $transport)
+    {
+        $this->transport = $transport;
+    }
+
+    public function __invoke(WeatherRequest $weatherRequest): WeatherResponse
+    {
+        # You could add data conversion, error handling, ... here. 
+        return WeatherResponse::fromRawArray(
+            ($this->transport)($weatherRequest)
+        );
+    }
+}
+```
+
+## Example Request Value Object
+```php
+<?php
+declare(strict_types=1);
+
+namespace Mage\OpenWeather\Http\Request;
+
+use Phpro\HttpTools\Request\RequestInterface;
+
+class WeatherRequest implements RequestInterface
+{
+
+    public function method(): string
+    {
+        return 'GET';
+    }
+
+    public function uri(): string
+    {
+        return 'data/2.5/weather?q={location}&APPID={appid}';
+    }
+
+    public function uriParameters(): array
+    {
+        return [
+            'location' => 'Antwerp,be',
+            'appid' => '...'
+        ];
+    }
+
+    public function body(): array
+    {
+        return [];
+    }
+}
+```
+
+## Example Response Value Object
+```php
+<?php
+declare(strict_types=1);
+
+namespace Mage\OpenWeather\Http\Response;
+
+class WeatherResponse
+{
+    private $type;
+
+    private $description;
+
+    public function __construct(string $type, string $description)
+    {
+        $this->type = trim($type);
+        $this->description = trim($description);
+    }
+
+    public static function fromRawArray(array $data): self
+    {
+        if (!isset($data['weather']) || !is_array($data['weather']) || !isset($data['weather'][0]['main'])) {
```

Comment:
IMO it's better to normalize the data and cast it into the correct type instead of throwing a real excetion here.
Unless it is a super blocking api call.

E.g. if the weather array is not available, you could fall back to 2 empty strings.
On the other hand: if weather is not available, maybe there is an "error" property you can as  better error message?